canary/webrtc-c/src/Peer.cpp

```diff
@@ -335,9 +379,10 @@ STATUS Peer::connect()
         SignalingMessage msg;
 
         MEMSET(&offerSDPInit, 0, SIZEOF(offerSDPInit));
-        CHK_STATUS(createOffer(this->pPeerConnection, &offerSDPInit));
+        DLOGI("Trickle ICE: %s", this->trickleIce ? "Enabled" : "Disabled");
+        offerSDPInit.useTrickleIce = this->trickleIce;
         CHK_STATUS(setLocalDescription(this->pPeerConnection, &offerSDPInit));
-
+        CHK_STATUS(createOffer(this->pPeerConnection, &offerSDPInit));
```

Comment:
Why are we creating the offer after setting local description?

Reply:
This is also done in the sample. It seems that it is done so to kick off `iceAgentStartGathering` as soon as possible which is done via `setLocalDescription`, and `setLocalDescription` does not use `&offerSDPInit` other than to verify that it is not null.

Reply:
In the JS at least, it is like this. Seeing it the other way around is really odd.
```
console.log('[VIEWER] Creating SDP offer');
await viewer.peerConnection.setLocalDescription(
    await viewer.peerConnection.createOffer({
        offerToReceiveAudio: true,
        offerToReceiveVideo: true,
    }),
);
```